components/director/internal/tenantfetcher/converter.go

```diff
@@ -0,0 +1,98 @@
+package tenantfetcher
+
+import (
+	"encoding/json"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type TenantFieldMapping struct {
+	NameField          string `envconfig:"default=name,APP_MAPPING_FIELD_NAME"`
+	IDField            string `envconfig:"default=id,APP_MAPPING_FIELD_ID"`
+	DiscriminatorField string `envconfig:"APP_MAPPING_FIELD_DISCRIMINATOR"`
+	DiscriminatorValue string `envconfig:"APP_MAPPING_VALUE_DISCRIMINATOR"`
+}
+
+type converter struct {
+	providerName string
+	fieldMapping TenantFieldMapping
+}
+
+func NewConverter(providerName string, fieldMapping TenantFieldMapping) *converter {
+	return &converter{
+		providerName: providerName,
+		fieldMapping: fieldMapping,
+	}
+}
+
+func (c converter) EventsToTenants(eventsType EventsType, events []Event) []model.BusinessTenantMappingInput {
+	var tenants []model.BusinessTenantMappingInput
+	for _, event := range events {
+		tenant, err := c.EventToTenant(eventsType, event)
+		if err != nil {
```

Comment:
we ignore an error (only log it), so we behave like a tolerant reader, but IMO we should mark that we failed.